rest/blip_api_test.go

```diff
@@ -4661,3 +4661,111 @@ func TestUnsubChanges(t *testing.T) {
 	_, found = btc.WaitForRev("doc2", resp.Rev)
 	assert.True(t, found)
 }
+
+func TestBlipGetCollections(t *testing.T) {
+
+	base.SetUpTestLogging(t, base.LevelDebug, base.KeyAll)
+
+	rt := NewRestTester(t, &RestTesterConfig{
+		guestEnabled: true,
+		DatabaseConfig: &DatabaseConfig{
+			DbConfig: DbConfig{
+				Scopes: ScopesConfig{
+					"fooScope": ScopeConfig{
+						Collections: map[string]CollectionConfig{
+							"fooCollection": {},
+						},
+					},
+				},
+			},
+		},
+	})
+	btc, err := NewBlipTesterClientOptsWithRT(t, rt, nil)
+	require.NoError(t, err)
+	defer btc.Close()
+
+	checkpointBody := db.Body{}
+	checkpointID1 := "checkpoint1"
+	dbInstance := db.Database{DatabaseContext: rt.GetDatabase()}
+	revID, err := dbInstance.PutSpecial(db.DocTypeLocal, db.CheckpointDocIDPrefix+checkpointID1, checkpointBody)
+	require.NoError(t, err)
+	checkpoint1RevID := "0-1"
+	require.Equal(t, checkpoint1RevID, revID)
+	testCases := []struct {
+		name        string
+		requestBody db.GetCollectionsRequestBody
+		resultBody  []db.Body
+		errorCode   string
+	}{
+		{
+			name: "noDocInDefaultCollection",
+			requestBody: db.GetCollectionsRequestBody{
+				CheckpointIDs: []string{"id"},
+				Collections:   []string{"_default._default"},
+			},
+			resultBody: []db.Body{nil},
+			errorCode:  "",
+		},
+		{
+			name: "mismatchedLengthOnInput",
+			requestBody: db.GetCollectionsRequestBody{
+				CheckpointIDs: []string{"id", "id2"},
+				Collections:   []string{"_default._default"},
+			},
+			resultBody: []db.Body{nil},
+			errorCode:  fmt.Sprintf("%d", http.StatusBadRequest),
+		},
+		{
+			name: "inDefaultCollection",
+			requestBody: db.GetCollectionsRequestBody{
+				CheckpointIDs: []string{checkpointID1},
+				Collections:   []string{"_default._default"},
+			},
+			resultBody: []db.Body{nil},
+			errorCode:  "",
+		},
+		{
+			name: "badScopeSpecificationEmptyString",
+			// bad scope specification - empty string
+			requestBody: db.GetCollectionsRequestBody{
+				CheckpointIDs: []string{checkpointID1},
+				Collections:   []string{""},
+			},
+			resultBody: []db.Body{nil},
+			errorCode:  fmt.Sprintf("%d", http.StatusBadRequest),
+		},
+		{
+			name: "inNonDefaultCollection",
+			requestBody: db.GetCollectionsRequestBody{
+				CheckpointIDs: []string{checkpointID1},
+				Collections:   []string{"fooScope.fooCollection"},
+			},
+			resultBody: []db.Body{db.Body{}},
+			errorCode:  "",
+		},
```

Comment:
Need a test case for the correct collection name, but an unseen checkpoint ID.

In this case we need to let CBL know they can use this collection but don't have a checkpoint to start from.

```suggestion
		{
			name: "unseenCheckpointNonDefaultCollection",
			requestBody: db.GetCollectionsRequestBody{
				CheckpointIDs: []string{"id"},
				Collections:   []string{"fooScope.fooCollection"},
			},
			resultBody: []db.Body{db.Body{}},
			errorCode:  "",
		},
```